libs/preallocate/src/main/java/org/elasticsearch/preallocate/Preallocate.java

```diff
@@ -58,12 +58,15 @@ public static void preallocate(final Path cacheFile, final long fileSize) throws
         }
     }
 
-    @SuppressForbidden(reason = "need access to toFile for RandomAccessFile")
+    @SuppressForbidden(reason = "need access to toFile for RandomAccessFile and mkdirs")
     private static void preallocate(final Path cacheFile, final long fileSize, final Preallocator prealloactor) throws IOException {
         boolean success = false;
         try {
             if (prealloactor.useNative()) {
-                try (NativeFileHandle openFile = prealloactor.open(cacheFile.toAbsolutePath().toString())) {
+                var absolutePath = cacheFile.toAbsolutePath();
+                var directory = absolutePath.getParent();
+                directory.toFile().mkdirs();
```

Comment:
nit (didn't notice this previously): this could be Files.createDirectories so you don't have to convert to File?

Reply:
Nice, I have the suspicion there was something to replace mkdirs but could not find it. Thanks for pointing this out, I will change it on main too.